sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -164,9 +194,34 @@ class RocksDB(
         loadedVersion = -1  // invalidate loaded data
         throw t
     }
+    if (enableChangelogCheckpointing && !readOnly) {
+      changelogWriter = Some(fileManager.getChangeLogWriter(version + 1))
+    }
     this
   }
 
+  private def replayChangelog(endVersion: Long): Unit = {
+    // Replay change log from the loaded version to the target version.
```

Comment:
Maybe move the comment before function start ?

Reply:
Moved to function comment.